app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,63 @@
+class ArticlesController < ApplicationController
+  before_action :set_article, only: [:edit, :update, :show, :destroy]
+  before_action :require_user
```

Comment:
instead of making this callback particular to each controller, this could have been made a callback in `ApplicationController` so if any controller inherits from this will have that callback applied by default.
If some controller doesn't require this then we can use `skip` callback